### PR TITLE
Debug direction arrow

### PR DESCRIPTION
**Pull request by freyinds**
Working implementation for now. Need to cache arrow model in the future.

### CODE REVIEW

File: commons/src/main/com/mbrlabs/mundus/commons/assets/AssetType.java [hunk 1]
```diff
@@ -31,6 +31,7 @@ public enum AssetType {
     MODEL("Model"),
     /** Terra file. Contains height data for terrains. */
     TERRAIN("Terrain"),
+    TERRAIN_LAYER("TL"),
```
Comment (Dgzt) [inline commons/src/main/com/mbrlabs/mundus/commons/assets/AssetType.java:34]: Why did you add this line into AssetType? As I see this enum type is not used.

File: commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java [hunk 2]
```diff
@@ -129,6 +137,28 @@ public void render(GameObject go) {
             instances.add(modelInstance);
         }
 
+        if (drawFacingArrow && go == selectedGameObject){
+            //TODO Cache arrow model
+            Vector3 origin = new Vector3();
```
Comment (Dgzt) [inline commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java:142]: You can get these object from [Pool](https://github.com/JamesTKhan/Mundus/blob/master/commons/src/main/com/mbrlabs/mundus/commons/utils/Pools.java) and you don't need to create new object in a render method.

File: commons/src/main/com/mbrlabs/mundus/commons/assets/AssetType.java (hunk 1 above)
Comment (freyinds) [reply commons/src/main/com/mbrlabs/mundus/commons/assets/AssetType.java:34]: Mistake will be removed on next commit.

File: commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java (hunk 2 above)
Comment (freyinds) [reply commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java]: Thanks for the Pool() tip! This PR is ready.

Comment (Dgzt) [reply commons/src/main/com/mbrlabs/mundus/commons/utils/DebugRenderer.java]: And don't forget to call Pools.vector3Pool.free(origin) and Pools.vector3Pool.free(facing) end of block (about line 157). This method add back the object to the pool and can be used the object again if call obtain method. If you don't call free method, like now, the GC will delete these object.

**Comment by JamesTKhan** [comment]
Looks like its on the right track. Two things:

1. It is not handling scaled objects well. Part of the issue is creating the Arrow using origin and facing values. Instead we should keep the origin and face values of the arrow at 0,0,0, and since Z+ is forward direction, facing is 0,0,arrowLength. By doing this, we can also get away with using the same arrow model all the time and do not need to cache it, since we just modify its position and rotation as needed. I attached a version with changes. Can you try applying the attached version on your branch and if it looks good, use this version? (NOTE: You will need to merge latest master branch in otherwise will see some errors)
[DebugRenderer.txt](https://github.com/JamesTKhan/Mundus/files/12457989/DebugRenderer.txt)

2. The setting for showing the arrow is being saved in preferences, but its not being set on application start up. Check out Editor.kt line 99 to see how to retrieve the preference and set it on the debugRenderer

```
        debugRenderer.isEnabled = globalPreferencesManager.getBoolean(MundusPreferencesManager.GLOB_BOOL_DEBUG_RENDERER_ON, false)
        debugRenderer.isAppearOnTop = globalPreferencesManager.getBoolean(MundusPreferencesManager.GLOB_BOOL_DEBUG_RENDERER_DEPTH_OFF, false)

```

**Comment by freyinds** [comment]
Looks good, only change was swap Math.max here: float scaleFactor = Math.min(camera.position.dst(worldPos) * 0.1f, cullableComponent.getRadius()); So that it properly scales with large scaled models.

**Comment by JamesTKhan** [comment]
> Looks good, only change was swap Math.max here: float scaleFactor = Math.min(camera.position.dst(worldPos) * 0.1f, cullableComponent.getRadius()); So that it properly scales with large scaled models. Happy to switch back to min if you all like it better.

One last issue and then I can merge as everything else looks good. The UI for the checkbox needs to be updated to reflect current setting. You probably just need to update it here: https://github.com/JamesTKhan/Mundus/blob/dd896acb7767e8595df44ea4101bdb5e249b48e7/editor/src/main/com/mbrlabs/mundus/editor/ui/modules/dialogs/tools/DebugRenderDialog.kt#L58

Right now, even though it saved and loaded the preference to show facing arrow, the UI is not checked on start up

image

**Review by JamesTKhan — APPROVED** [review]
Looks good. Thank you!
